Add documentation of the vulnerability detector tests (#820)
wazuh/wazuh-qa: tests/integration/test_vulnerability_detector/readme.md

value of `enabled` tag in `ossec.conf` and checking the result in `ossec.log`.

- **[test_providers_os](./doc/test_providers/test_providers_os.md)**: Tests that are based on modifying the value
of `os` tag in `ossec.conf`, setting different os for Canonical and Debian and checking the result in `ossec.log`.

- **[test_providers_update_from_year](./doc/test_providers/test_providers_update_from_year.md)**: Tests that are based
on modifying the value of `update_from_year` tag in `ossec.conf`, setting different years and checking the result in
`ossec.log`.

- **[test_providers_update_interval](./doc/test_providers/test_providers_update_interval.md)**: Tests that are based
on modifying the value of `interval` tag in `ossec.conf`, setting different time intervals and units, and checking the
result in `ossec.log`.

---

### Tier 2

#### Test feeds - GENERIC

- **[test_invalid_type_custom_feeds](./doc/test_feeds/generic/test_invalid_type_custom_feeds.md)**: Tests that are
based on importing files of several different types (`.mp3`, `.jpg`, `.pdf` ...) as custom feed, and observing the
response of vulnerability detector.

> - **[test_validate_feed_content](./doc/test_feeds/generic/test_validate_feed_content.md)**: Tests that are based on
> downloading the feeds of all providers and verifying that the format of each feed is as expected and that the
> content of the feed is also `XML` or `JSON` parseable.

Reviewer: ```suggestion
- **[test_validate_feed_content](./doc/test_feeds/generic/test_validate_feed_content.md)**: Tests that download the feeds of all providers, verify the format of each feed is as expected and their content is also `XML` or `JSON` parseable.
```

#### Test feeds - RHEL

> - **[test_extra_fields_redhat_feed](./doc/test_feeds/redhat/test_extra_fields_redhat_feed.md)**: Set of tests that are
> based on checking the behavior of vulnerability detector when there is any extra field in the feed.

Reviewer: ```suggestion
- **[test_extra_fields_redhat_feed](./doc/test_feeds/redhat/test_extra_fields_redhat_feed.md)**: Set of tests that check the behavior of vulnerability detector when there is any extra field in the feed.
```

> - **[test_invalid_syntax_redhat_feed](./doc/test_feeds/redhat/test_invalid_syntax_redhat_feed.md)**: Set of tests that
> are based on checking the behavior of vulnerability detector when the feed has some kind of syntactic error: it doesn't
> have a closing field, it's missing some character ...

Reviewer: ```suggestion
- **[test_invalid_syntax_redhat_feed](./doc/test_feeds/redhat/test_invalid_syntax_redhat_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when the feed has some kind of syntactic error.
```

> - **[test_invalid_values_redhat_feed](./doc/test_feeds/redhat/test_invalid_values_redhat_feed.md)**: Set of tests that
> are based on checking the behavior of vulnerability detector when the value of a field is not correct: invalid type,
> strange characters ...

Reviewer: ```suggestion
- **[test_invalid_values_redhat_feed](./doc/test_feeds/redhat/test_invalid_values_redhat_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when the value of a field is incorrect.
```

> - **[test_missing_fields_redhat_feed](./doc/test_feeds/redhat/test_missing_fields_redhat_feed.md)**: Set of tests that
> are based on checking the behavior of vulnerability detector when any field is missing from the feed.

Reviewer: ```suggestion
- **[test_missing_fields_redhat_feed](./doc/test_feeds/redhat/test_missing_fields_redhat_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when any field is missing from the feed.
```

#### Test feeds - CANONICAL

> - **[test_extra_tags_canonical_feed](./doc/test_feeds/canonical/test_extra_tags_canonical_feed.md)**:  Set of tests
> that are based on checking the behavior of vulnerability detector when there is any extra tag in the feed.

Reviewer: ```suggestion
- **[test_extra_tags_canonical_feed](./doc/test_feeds/canonical/test_extra_tags_canonical_feed.md)**: Set of tests
that check the behavior of Vulnerability Detector when there is any extra tag in the feed.
```

> - **[test_invalid_syntax_canonical_feed](./doc/test_feeds/canonical/test_invalid_syntax_canonical_feed.md)**: Set of
> tests that are based on checking the behavior of vulnerability detector when the feed has some kind of syntactic error:
> it doesn't have a closing tag, it's missing some character ...

Reviewer: ```suggestion
- **[test_invalid_syntax_canonical_feed](./doc/test_feeds/canonical/test_invalid_syntax_canonical_feed.md)**: Set of
tests that check the behavior of Vulnerability Detector when the feed has some kind of syntactic error.
```

> - **[test_invalid_values_canonical_feed](./doc/test_feeds/canonical/test_invalid_values_canonical_feed.md)**: Set of
> tests that are based on checking the behavior of vulnerability detector when the value of a tag is not correct:
> invalid type, strange characters ...

Reviewer: ```suggestion
- **[test_invalid_values_canonical_feed](./doc/test_feeds/canonical/test_invalid_values_canonical_feed.md)**: Set of
tests that check the behavior of Vulnerability Detector when the value of a tag is incorrect.
```

> - **[test_missing_tags_canonical_feed](./doc/test_feeds/canonical/test_missing_tags_canonical_feed.md)**: Set of tests
> that are based on checking the behavior of vulnerability detector when any tag is missing from the feed.

Reviewer: ```suggestion
- **[test_missing_tags_canonical_feed](./doc/test_feeds/canonical/test_missing_tags_canonical_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when any tag is missing from the feed.
```

#### Test feeds - DEBIAN

> - **[test_extra_tags_debian_feed](./doc/test_feeds/debian/test_extra_tags_debian_feed.md)**: Set of tests that are
> based on checking the behavior of vulnerability detector when there is any extra tag in the feed.

Reviewer: ```suggestion
- **[test_extra_tags_debian_feed](./doc/test_feeds/debian/test_extra_tags_debian_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when there is any extra tag in the feed.
```

> - **[test_invalid_syntax_debian_feed](./doc/test_feeds/debian/test_invalid_syntax_debian_feed.md)**: Set of tests that
> are based on checking the behavior of vulnerability detector when the feed has some kind of syntactic error: it doesn't
> have a closing tag, it's missing some character ...

Reviewer: ```suggestion
- **[test_invalid_syntax_debian_feed](./doc/test_feeds/debian/test_invalid_syntax_debian_feed.md)**: Set of tests check the behavior of Vulnerability Detector when the feed has some kind of syntactic error.
```

> - **[test_invalid_values_debian_feed](./doc/test_feeds/debian/test_invalid_values_debian_feed.md)**: Set of tests that
> are based on checking the behavior of vulnerability detector when the value of a tag is not correct: invalid type,
> strange characters ...

Reviewer: ```suggestion
- **[test_invalid_values_debian_feed](./doc/test_feeds/debian/test_invalid_values_debian_feed.md)**: Set of tests that check the behavior of vulnerability detector when the value of a tag is not correct.
```

> - **[test_missing_tags_debian_feed](./doc/test_feeds/debian/test_missing_tags_debian_feed.md)**: Set of tests that are
> based on checking the behavior of vulnerability detector when any tag is missing from the feed.

Reviewer: ```suggestion
- **[test_missing_tags_debian_feed](./doc/test_feeds/debian/test_missing_tags_debian_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when any tag is missing from the feed.
```
